source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/clients/versioncontrol/soapextensions/Workspace.java

```diff
@@ -8770,4 +8773,32 @@ private boolean acesMatch(
 
         return true;
     }
+
+    public static String computeNewWorkspaceName(final String baseName, final Workspace[] existingWorkspaces) {
+        /*
+         * Given a base name like "machinea", we are trying to compute a name
+         * that doesn't already exist in the exisstingWorkspaces array. If
+         * "machinea" is taken, we try "machinea_1", then "machinea_2", etc.
+         */
+
+        String candidateName = baseName;
+        boolean keepGoing = true;
+        int ix = 0;
+        while (keepGoing) {
+            boolean matched = false;
+            for (int i = 0; !matched && i < existingWorkspaces.length; i++) {
+                matched = existingWorkspaces[i].getName().equalsIgnoreCase(candidateName);
+                if (matched) {
+                    break;
+                }
+            }
+            if (!matched) {
+                keepGoing = false;
+            } else {
+                final String messageFormat = Messages.getString("Workspace.DefaultNewWorkspaceNameFormat"); //$NON-NLS-1$
+                candidateName = MessageFormat.format(messageFormat, baseName, Integer.toString(++ix));
+            }
+        }
+        return candidateName;
+    }
```

Comment:
Could you please add unit testing for computeNewWorkspaceName()?  With this suggestion, it all looks good to me.